gr-blocks/python/blocks/msg_pair_to_var.py

```diff
@@ -24,13 +27,17 @@ def __init__(self, callback):
         self.set_msg_handler(pmt.intern("inpair"), self.msg_handler)
 
     def msg_handler(self, msg):
-        try:
-            new_val = pmt.to_python(pmt.cdr(msg))
+        if not pmt.is_pair(msg) or pmt.is_dict(msg) or pmt.is_pdu(msg):
+            gr.log.warn("Input message %s is not a simple pair, dropping" % repr(msg))
```

Comment:
dict and pdu are allowed?

Reply:
This block expects a simple pair, (usually a key-value pair?) so the conditional will drop the message if it is not a `pair`, or if its a `dict`, or if its a `pdu`.

Both `dict` and `pdu` type PMTs pass `is_pair()` so they are called out explicitly to be dropped.

Reply:
OK, I missed the `not` and read it wrong.

Reply:
All good. It's debatable if PDUs should be dropped. I think they should though, given the intention of the block.